components/benefit_sponsors/app/controllers/benefit_sponsors/profiles/registrations_controller.rb

```diff
@@ -79,7 +79,9 @@ def counties_for_zip_code
       private
 
       def profile_type
-        @profile_type = params[:profile_type] || params[:agency][:profile_type] || @agency.profile_type
+        valid_profile_types = %w[benefit_sponsor broker_agency general_agency].freeze
```

Comment:
@ZASMan minor comment(to dry) since you added a constant VALID_PROFILE_TYPES in components/benefit_sponsors/app/models/benefit_sponsors/services/new_profile_registration_service.rb, you don't need to create this variable again here.